Add docs for sso-team-id to the Teams API (#2210)
hashicorp/terraform-website: content/cloud-docs/users-teams-organizations/single-sign-on/index.mdx

## Managing Team Membership Through SSO

Terraform Cloud can automatically add users to teams based on their SAML assertion, so you can manage team membership in your directory service. To do this, you must specify the `MemberOf` SAML attribute, and make sure the AttributeStatement in the SAMLResponse contains a list of AttributeValue items in the correct format (i.e., comma-separated list of team names or SSO team IDs).

If the SAML assertion includes a `MemberOf` attribute, users logging in via SSO are automatically added to the teams included in their assertion and removed from any teams that _aren't_ included in their assertion. This overrides any manually set team memberships; whenever the user logs in, their team membership is adjusted to match their SAML assertion.

Any team names that don't match existing teams are ignored; Terraform Cloud will not automatically create new teams. Terraform Cloud expects the team names in the `MemberOf` SAML attribute to exactly match its own team names. This match is case sensitive. It is not possible to assign users to the `owners` team through this attribute.

> In some circumstances the `MemberOf` SAML attribute may not be human readable, due to restrictions in your SSO provider for example. In this case, in order to maintain human readalbe team names in Terraform Cloud, you may set an SSO Team ID for the team which matches the SAML attribute instead.

Reviewer: I thought to myself this morning that it might be nice to have a paragraph explaining _why_ SSO Team IDs are a thing, so I added a blurb here. I thought I could get it in before you re-reviewed @laurapacilio but no luck :(

Would you mind proofing this line to make sure you still approve?

> In some circumstances the `MemberOf` SAML attribute may not be human readable, due to restrictions in your SSO provider for example. In this case, in order to maintain human readalbe team names in Terraform Cloud, you may set an SSO Team ID for the team which matches the SAML attribute instead.

Reviewer 2: ```suggestion
The `MemberOf` SAML attribute may not be human readable. To maintain human readable team names in Terraform Cloud, you can set an SSO Team ID that matches the SAML attribute instead.
```

## Managing Team Membership Through SSO

Terraform Cloud can automatically add users to teams based on their SAML assertion, so you can manage team membership in your directory service.

Team membership mapping is controlled with the "Enable team management" toggle on the SSO configuration page of your Organization settings.

![Screenshot: the Terraform Cloud SAML team membership toggle](/img/docs/saml-team-membership-cloud.png)

> When enabled, you may configure which SAML attribute in the SAMLResponse will control team membership. This defaults to the `MemberOf` attribute. The expected format of the corresponding AttributeValue in the SAMLResponse is a either a string containing a comma-separated list of teams, or separate AttributeValue items specifying teams.

Reviewer: Sure! Updated

When team membership management is enabled, users logging in via SAML are automatically added to the teams included in their assertion, and automatically removed from any teams that _aren't_ included in their assertion. This overrides any manually set team memberships; whenever the user logs in, their team membership is adjusted to match their SAML assertion.

Any team names that don't match existing teams are ignored; Terraform Cloud will not automatically create new teams. If the chosen SAML attribute is not provided in the SAMLResponse, users are assigned to a default team named `sso` and are not removed from any existing teams.

It is not possible to assign users to the `owners` team through this attribute.

## Team Names and SSO Team IDs

> Terraform Cloud expects the team names in the team membership SAML attribute to exactly match its own team names, or its configured SSO Team IDs. This match is case sensitive.

Reviewer 2: ```suggestion
Terraform Cloud expects the team names in the team membership SAML attribute to exactly match its own team names or its configured SSO Team IDs. This match is case sensitive.
```
